cocotb/binary.py

```diff
@@ -237,12 +237,12 @@ def _adjust_unsigned(self, x):
             else:
                 rv = '0' * (self._n_bits - l) + x
         elif l > self._n_bits:
-            print("WARNING: truncating value to match requested number of bits "
-                  "(%d -> %d)" % (l, self._n_bits))
             if self.big_endian:
                 rv = x[l - self._n_bits:]
             else:
                 rv = x[:l - self._n_bits]
+            print("WARNING: %d-bit value requested, truncating value %s (%d bits) to %s"
+                  % (self._n_bits, x, l, rv))
```

Comment:
Let's switch to using `warnings.warn` while we're here.

Reply:
Switching to warnings will cause failures in CI when adding tests for this feature. Or at least has in the past.

Reply:
@ktbarrett I have covered some in ``tests/pytest/test_binary_value.py``, let's see how many more there are.

Reply:
Hey, passes.